syft/file/digest.go

```diff
@@ -1,6 +1,62 @@
+// nolint:gosec
 package file
 
+import (
+	"crypto/md5"
+	"crypto/sha1"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"hash"
+	"io"
+	"os"
+
+	"github.com/anchore/syft/internal/log"
+)
+
+type HashAlgorithm string
```

Comment:
we should be able to use the crypto.Hash object to derive the name --there is already a helper function in the digest cataloger for this